web/styles/compose.css

```diff
@@ -1588,6 +1588,51 @@ textarea.new_message_textarea {
     }
 }
 
+.saved_snippets-dropdown-list-container {
+    width: 250px;
+
+    .dropdown-list .dropdown-list-item-common-styles {
+        padding: 5px 10px;
+        display: flex;
+        flex-direction: column;
+
+        .dropdown-list-item-name {
+            line-height: 15px;
+        }
+
+        .dropdown-list-item-description {
+            white-space: nowrap;
+            font-weight: 400;
+            font-size: 13px;
+            opacity: 0.8;
+            padding: 0;
+            text-overflow: ellipsis;
+            overflow: hidden;
+        }
+
+        .dropdown-list-bold-selected {
+            font-weight: 500;
+            max-width: 210px;
+            display: inline-block;
+            text-overflow: ellipsis;
+            overflow: hidden;
+            white-space: nowrap;
+        }
+    }
+}
+
+#add-new-saved-snippet-modal {
+    & .title {
+        width: 97%;
+        margin-bottom: 20px;
+    }
+
+    & .content {
```

Comment:
These would really benefit from better CSS class names, like `saved-snippet-title`. Generally you want every class name to be globally unique in the app, since it makes refactoring way way easier.

Reply:
``` diff
diff --git a/web/src/saved_snippets_ui.ts b/web/src/saved_snippets_ui.ts
index 86472fe2ed..a08e7f939d 100644
--- a/web/src/saved_snippets_ui.ts
+++ b/web/src/saved_snippets_ui.ts
@@ -18,16 +18,16 @@ import type {StateData} from "./state_data";
 let saved_snippet_dropdown_widget: dropdown_widget.DropdownWidget;
 
 function submit_create_saved_snippet_form(): void {
-    const title = $<HTMLInputElement>("#add-new-saved-snippet-modal .title").val()?.trim();
-    const content = $<HTMLInputElement>("#add-new-saved-snippet-modal .content").val()?.trim();
+    const title = $<HTMLInputElement>("#add-new-saved-snippet-modal .saved-snippet-title").val()?.trim();
+    const content = $<HTMLInputElement>("#add-new-saved-snippet-modal .saved-snippet-content").val()?.trim();
     if (title && content) {
         dialog_widget.submit_api_request(channel.post, "/json/saved_snippets", {title, content});
     }
 }
 
 function update_submit_button_state(): void {
-    const title = $<HTMLInputElement>("#add-new-saved-snippet-modal .title").val()?.trim();
-    const content = $<HTMLInputElement>("#add-new-saved-snippet-modal .content").val()?.trim();
+    const title = $<HTMLInputElement>("#add-new-saved-snippet-modal .saved-snippet-title").val()?.trim();
+    const content = $<HTMLInputElement>("#add-new-saved-snippet-modal .saved-snippet-content").val()?.trim();
     const $submit_button = $("#add-new-saved-snippet-modal .dialog_submit_button");
 
     $submit_button.prop("disabled", true);
@@ -89,7 +89,7 @@ function item_click_callback(
             form_id: "add-new-saved-snippet-form",
             update_submit_disabled_state_on_change: true,
             on_click: submit_create_saved_snippet_form,
-            on_shown: () => $("#saved-snippet-title").trigger("focus"),
+            on_shown: () => $("#add-saved-snippet-title").trigger("focus"),
             post_render: saved_snippet_modal_post_render,
         });
     } else {
diff --git a/web/styles/compose.css b/web/styles/compose.css
index 1fff79fbe5..1ec95b3081 100644
--- a/web/styles/compose.css
+++ b/web/styles/compose.css
@@ -1622,12 +1622,12 @@ textarea.new_message_textarea {
 }
 
 #add-new-saved-snippet-modal {
-    & .title {
+    & .saved-snippet-title {
         width: 97%;
         margin-bottom: 20px;
     }
 
-    & .content {
+    & .saved-snippet-content {
         width: 97%;
         resize: vertical;
     }
diff --git a/web/templates/add_saved_snippet_modal.hbs b/web/templates/add_saved_snippet_modal.hbs
index 7aa43f3bf5..75e73ef0b5 100644
--- a/web/templates/add_saved_snippet_modal.hbs
+++ b/web/templates/add_saved_snippet_modal.hbs
@@ -1,8 +1,8 @@
 <div id="add-new-saved-snippet-modal" class="new-style">
     <form id="add-new-saved-snippet-form" class="new-style">
         <label for="title" class="modal-field-label">{{t "Title" }}</label>
-        <input id="saved-snippet-title" type="text" name="title" class="modal_text_input title" value="" autocomplete="off" spellcheck="false" autofocus="autofocus"/>
+        <input id="new-saved-snippet-title" type="text" name="title" class="modal_text_input saved-snippet-title" value="" autocomplete="off" spellcheck="false" autofocus="autofocus"/>
         <div>{{t "Content" }}</div>
-        <textarea class="content settings_textarea" rows="4"></textarea>
+        <textarea class="settings_textarea saved-snippet-content" rows="4"></textarea>
     </form>
 </div>
```